src/configobj/__init__.py

```diff
@@ -1032,18 +1033,21 @@ class ConfigObj(Section):
     # TODO: also support inline comments (needs dynamic compiling of the regex below)
     COMMENT_MARKERS = ['#']
 
-    _keyword = re.compile(r'''^ # line start
-        (\s*)                   # indentation
-        (                       # keyword
-            (?:".*?")|          # double quotes
-            (?:'.*?')|          # single quotes
-            (?:[^'"=].*?)       # no quotes
-        )
-        \s*=\s*                 # divider
-        (.*)                    # value (including list values and comments)
-        $   # line end
-        ''',
-        re.VERBOSE)
+
+    @property
+    def _keyword(self):
+        return re.compile(r'''^ # line start
+            (\s*)                   # indentation
+            (                       # keyword
+                (?:".*?")|          # double quotes
+                (?:'.*?')|          # single quotes
+                (?:[^'"{0}].*?)     # no quotes
+            )
+            \s*[{0}]\s*             # dividers
+            (.*)                    # value (including list values and comments)
+            $   # line end
+            '''.format(''.join(self._dividers)),
+            re.VERBOSE)
```

Comment:
We should probably implement some caching here.  There's only one place that ``self._keyword`` is called, but it gets called a LOT.  By writing the property this way, we'll end up re-compiling the regex on almost every single line of the config file being parsed.  Admittedly, I haven't done any profiling, so it's possible that this is just something that seems bad but wouldn't matter in practice.

With that being said, it's an easy enough fix.  Instead of

```
return re.compile(...)
```

We can just say

```
if not hasattr(self, '__keyword'):
    self.__keyword = re.compile(...)
return self.__keyword
```

That will ensure that the regex only gets compiled the first time.